nixosTests.ly: adapt test to new user selection login

phanirithvij opened the pull request:
ZHF: https://github.com/NixOS/nixpkgs/issues/457852

Hydra: https://hydra.nixos.org/build/311255910

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

phanirithvij committed:
nixosTests.ly: adapt test to new user selection login

Signed-off-by: phanirithvij <phanirithvij2000@gmail.com>

wolfgangwalther commented:
Happy to do merges for already reviewed and approved PRs, but no cycles for actual review work right now, sorry!

Vonfry reviewed (approved):
Diff LGTM

phanirithvij commented:
Vonfry is the service author for ly.

phanirithvij commented:
@drupol it seems it was not due to my changes https://github.com/NixOS/nixpkgs/pull/427541#issuecomment-3111046240

phanirithvij committed:
nixosTests.ly: fix aarch64-linux test

It was failing with xserver crashing, full logs
https://github.com/NixOS/nixpkgs/pull/427541#issuecomment-3111046240

Co-Authored-By: Markus Wamser <wamserma@users.noreply.github.com>
Signed-off-by: phanirithvij <phanirithvij2000@gmail.com>

illdefined commented:
I tried this on `x86_64-linux` and both the Wayland and X variants appear to work, Sway however complains about the lack of hardware rendering:

```
machineNoX11 # [   39.108042] xsession[978]: 00:00:00.366 [ERROR] [wlr] [render/egl.c:334] Software rendering detected, please use the WLR_RENDERER_ALLOW_SOFTWARE environment variable to proceed
machineNoX11 # [   39.110573] xsession[978]: 00:00:00.369 [ERROR] [wlr] [render/egl.c:611] Failed to initialize EGL context
machineNoX11 # [   39.112389] xsession[978]: 00:00:00.369 [ERROR] [wlr] [render/gles2/renderer.c:499] Could not initialize EGL
machineNoX11 # [   39.346539] xsession[978]: 00:00:00.605 [ERROR] [wlr] [render/vulkan/renderer.c:2516] Could not match drm and vulkan device
```

A screenshot is created successfully nonetheless.

phanirithvij commented on nixos/tests/ly.nix:
```diff
@@ -22,6 +22,13 @@ in
+        ACTION=="add", SUBSYSTEM=="drm", KERNEL=="card0", TAG+="systemd"
+      '';
+      systemd.services.display-manager.wants = [ "dev-dri-card0.device" ];
+      systemd.services.display-manager.after = [ "dev-dri-card0.device" ];
```
It seems to be an unresolved bug that's been reported quite a while ago. https://github.com/systemd/systemd/issues/25408

Maybe there are many other nixosTests running xorg that face this issue.

I hit it only once out of a hundred on x86_64, when I ran the ly test in a loop. But it consistently happens on aarch64.

wamserma reviewed (approved):
LGTM
